graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/migration/CassandraMigrator.java

```diff
@@ -0,0 +1,174 @@
+          throw new AssertionError("Unexpected target " + entity.getTarget());
+      }
+    }
+    if (!errors.isEmpty()) {
+      throw GraphqlErrorException.newErrorException()
+          .message(
+              String.format(
+                  "The schema you provided can't be mapped to the current CQL data model."
+                      + "Consider using a different migration strategy (current: %s). "
+                      + "See details in `extensions.migrationErrors` below.",
+                  strategy))
+          .extensions(ImmutableMap.of("migrationErrors", errors))
+          .build();
+    }
+    return ImmutableList.copyOf(queries);
+  }
+
+  private <T extends SchemaEntity> void compute(
+      T expected,
+      T actual,
+      BiFunction<T, T, List<Difference>> comparator,
+      BiFunction<T, DataStore, MigrationQuery> createBuilder,
+      BiFunction<T, DataStore, MigrationQuery> dropBuilder,
+      CassandraSchemaHelper.AddColumnBuilder<T> addColumnBuilder,
+      String entityType,
+      Set<MigrationQuery> queries,
+      List<String> errors) {
+    if (strategy == MigrationStrategy.DROP_AND_RECREATE_ALL) {
+      // We don't need to compare the schema, just override everything.
+      queries.add(dropBuilder.apply(expected, dataStore));
+      queries.add(createBuilder.apply(expected, dataStore));
+    } else if (actual == null) {
+      // The table/UDT does not exist
+      switch (strategy) {
+        case USE_EXISTING:
+          errors.add(String.format("Missing %s %s", entityType, expected.name()));
+          break;
+        case ADD_MISSING_TABLES:
+        case ADD_MISSING_TABLES_AND_COLUMNS:
+        case DROP_AND_RECREATE_IF_MISMATCH:
+          queries.add(createBuilder.apply(expected, dataStore));
+          break;
+        default:
+          throw new AssertionError("Unexpected strategy " + strategy);
+      }
+    } else {
+      // The table/UDT exists, compare with our model
+      List<Difference> differences = comparator.apply(expected, actual);
+      if (!differences.isEmpty()) {
+        switch (strategy) {
+          case USE_EXISTING:
+          case ADD_MISSING_TABLES:
+            errors.addAll(
+                differences.stream().map(Difference::toString).collect(Collectors.toList()));
+            break;
+          case ADD_MISSING_TABLES_AND_COLUMNS:
+            List<Difference> blockers =
+                differences.stream().filter(d -> !isAddableColumn(d)).collect(Collectors.toList());
+            if (blockers.isEmpty()) {
+              for (Difference difference : differences) {
+                assert isAddableColumn(difference);
+                queries.add(addColumnBuilder.build(expected, difference.getColumn(), dataStore));
+              }
+            } else {
+              errors.addAll(
+                  blockers.stream().map(Difference::toString).collect(Collectors.toList()));
+            }
+            break;
+          case DROP_AND_RECREATE_IF_MISMATCH:
```

Comment:
Can we refactor it to simplify Ifs and switch? Right now it is hard to read because of enclosed conditionals.
Also, how this `DROP_AND_RECREATE_IF_MISMATCH` is different from `DROP_AND_RECREATE_ALL`? I see that they execute the same logic. Can we merge those together?

Reply:
Yeah the logic is a bit convoluted, I'll see how I can improve it.

`DROP_AND_RECREATE_ALL` drops and recreates all tables no matter what.

`DROP_AND_RECREATE_IF_MISMATCH` only recreates the tables that have changed. For example, let's say you have an existing schema with a `User` type and some associated data. You're working on a v2 that adds an `Address` table. This strategy allows you to iterate on `Address` with drop-and-recreate (which is the most convenient approach for completely new types), without losing your existing data for `User`.

Reply:
I've changed to a single switch statement, it should be easier to follow now.